lib/Controller/UsersController.php

```diff
@@ -425,29 +484,10 @@ public function create($username, $password, array $groups= [], $email='') {
 			 */
 			if ($email !== '') {
 				$user->setEMailAddress($email);
-
-				// data for the mail template
-				$mailData = [
-					'username' => $username,
-					'url' => $this->urlGenerator->getAbsoluteURL('/')
-				];
-
-				$mail = new TemplateResponse('user_management', 'new_user/email-html', $mailData, 'blank');
-				$mailContent = $mail->render();
-
-				$mail = new TemplateResponse('user_management', 'new_user/email-plain_text', $mailData, 'blank');
-				$plainTextMailContent = $mail->render();
-
-				$subject = $this->l10n->t('Your %s account was created', [$this->defaults->getName()]);
-
 				try {
-					$message = $this->mailer->createMessage();
-					$message->setTo([$email => $username]);
-					$message->setSubject($subject);
-					$message->setHtmlBody($mailContent);
-					$message->setPlainBody($plainTextMailContent);
-					$message->setFrom([$this->fromMailAddress => $this->defaults->getName()]);
-					$this->mailer->send($message);
+					$this->generateTokenAndSendMail($username, $email);
+					//The value 43200 = 60*60*12 = 1/2 day
+					$this->config->setAppValue('user_management', 'token_expire_time', 43200);
```

Comment:
are you changing the app's global configuration every time a new token is generated ?!

Reply:
Removed.